tests/test_server.py

```diff
@@ -384,9 +384,12 @@ async def test_get_resource_for_agent(server_multi, client_multi, environment_mu
     assert result.result["model"]["released"]
     assert result.result["model"]["result"] == "deploying"
 
-    result = await aclient.get_resources_for_agent(environment_multi, "vm1.dev.inmanta.com")
-    assert result.code == 200
-    assert len(result.result["resources"]) == 3
+    async def wait_for_session() -> bool:
+        result = await aclient.get_resources_for_agent(environment_multi, "vm1.dev.inmanta.com")
+        assert len(result.result["resources"]) == 3
+        return result.code == 200
```

Comment:
We want to cover the scenario where a 409 status code is returned, so we have to include it in the returned value.
```suggestion
        return result.code == 200 and len(result.result["resources"]) == 3
```